classes/genimage.bbclass

```diff
@@ -103,6 +105,18 @@ do_genimage[depends] += "${@'${GENIMAGE_ROOTFS_IMAGE}:do_image_complete' if '${G
 GENIMAGE_CREATE_BMAP ?= "0"
 do_genimage[depends] += "${@'bmap-tools-native:do_populate_sysroot' if d.getVar('GENIMAGE_CREATE_BMAP') == '1' else ''}"
 
+GENIMAGE_COMPRESSION ??= "none"
+
+GENIMAGE_COMPRESS_DEPENDS[none] = ""
+GENIMAGE_COMPRESS_DEPENDS[gzip] = "pigz-native:do_populate_sysroot"
+GENIMAGE_COMPRESS_DEPENDS[xz]   = "xz-native:do_populate_sysroot"
+do_genimage[depends] += "${@d.getVarFlag('GENIMAGE_COMPRESS_DEPENDS', ${GENIMAGE_COMPRESSION})}"
```

Comment:
I gave this a try to replace the `exec-post = "gzip -k $IMAGEOUTFILE"` I have and it fails to parse for me because `${GENIMAGE_COMPRESSION}` is not within quotes:

> 
>   File "Var <do_genimage[depends]>", line 1, in <module>
>   bb.data_smart.ExpansionError: Failure expanding variable do_genimage[depends], expression was  `${@':do_image_complete' if '' else ''}` `${@'bmap-tools-native:do_populate_sysroot' if d.getVar('GENIMAGE_CREATE_BMAP') == '1' else ''}` `${@d.getVarFlag('GENIMAGE_COMPRESS_DEPENDS', gzip)}` `virtual/fakeroot-native:do_populate_sysroot`        `virtual/bootloader:do_deploy`    `myrootfs:do_deploy`    which triggered exception `NameError: name 'gzip' is not defined`
>   The variable dependency chain for the failure is: `do_genimage[depends]`
>